src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,39 @@
+import { type ReactNode, useEffect } from 'react';
+
+interface ModalProps {
+  onClose: () => void;
+  children: ReactNode;
+}
+
+export default function Modal({ onClose, children }: ModalProps) {
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        onClose();
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [onClose]);
+
+  return (
+    <div
+      onClick={onClose}
```

Comment:
To avoid closing the modal window when you try to select something in the modal window and accidentally move the mouse outside of it, you can try the following
```suggestion
      onMouseDown={onClose}
```